substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -2229,3 +2196,25 @@ impl<T: Config> ChangeNode<LocationOf<T>, PubConfigOf<T>, InterfaceOf<T>, Serial
         }
     }
 }
+
+impl<T: Config> PublicIpModifier for Pallet<T> {
+    fn ip_removed(ip: &PublicIP) {
+        if let Some(mut deployment) = Deployments::<T>::get(ip.contract_id) {
+            // Todo remove public ip from deployment and capacity contract
+            let cr_contract = Contracts::<T>::get(deployment.capacity_reservation_id);
+            if let Some(mut cr_contract) = cr_contract {
+                // free the public ips requested by the contract
+                if deployment.public_ips > 0 {
+                    match Self::_free_ip(ip.contract_id, &mut deployment, &mut cr_contract) {
+                        Ok(_) => (),
+                        Err(e) => {
+                            log::error!("error while freeing ips: {:?}", e);
+                        }
+                    }
```

Comment:
Since you do nothing in the Ok case, this match statement can be replaced by `if let Err(e) = Self::_free_ip(...) { log::error!(...); }`